pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+                self.assertEqual(body['importer_' + key], importers[0][key])
+
+
+class SyncValidFeedTestCase(_BaseTestCase):
+    """Test creating repository with valid feed.
+
+    Create two repositories wih valid feeds, one with valid and one with
+    invalid query. Check that sync finish succesfully in both of them.
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an puppet repositories with a valid feed and sync it."""
+        super(SyncValidFeedTestCase, cls).setUpClass()
+        bodies = tuple((_gen_puppet_repo_body() for i in range(2)))
+        bodies[0]['importer_config']['feed'] = _VALID_PUPPET_FEED
+        bodies[0]['importer_config']['queries'] = [_VALID_PUPPET_QUERY]
+        bodies[1]['importer_config']['feed'] = _VALID_PUPPET_FEED
+        bodies[1]['importer_config']['queries'] = [_INVALID_PUPPET_QUERY]
+        cls.attrs_iter = tuple((
+            create_repository(cls.cfg, body) for body in bodies
+        ))
+        cls.sync_repo = []  # raw responses
+        cls.task_bodies = tuple((chain.from_iterable(  # response bodies
+            poll_spawned_tasks(cls.cfg, call_report)
+            for call_report in (_sync_repo(
+                cls.cfg,
+                attr_iter['_href'],
+                cls.sync_repo
+            ) for attr_iter in cls.attrs_iter)
+        )))
+
+    def test_start_sync_code(self):
+        """Assert the call to sync each repository returns an HTTP 202."""
+        for i in range(2):
+            with self.subTest(i=i):
+                self.assertEqual(self.sync_repo[i].status_code, 202)
+
+    def test_task_error(self):
+        """Assert each task's "error" field is null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertEqual(task_body['error'], None)
+
+    def test_task_traceback(self):
+        """Assert each task's "traceback" field is null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertEqual(task_body['traceback'], None)
+
+    def test_task_progress_report(self):
+        """Assert no task's progress report contains error details."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertEqual(
+                    task_body['progress_report']['puppet_importer']['metadata']['error_message'],  # noqa pylint:disable=line-too-long
+                    None
+                )
+
+
+class SyncInvalidFeedTestCase(_BaseTestCase):
+    """If an invalid feed is given, the sync completes with reported errors."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create puppet repository with an invalid feed and sync it."""
+        super(SyncInvalidFeedTestCase, cls).setUpClass()
+        body = _gen_puppet_repo_body()
+        body['importer_config']['feed'] = ''.join(['http://', uuid4()])
```

Comment:
How about dropping `''.join()`, like so:

```python
body['importer_config']['feed'] = 'http://' + uuid4()
```

That said, is there any reason to use `http://` at all?